Makefile

```diff
@@ -187,4 +218,33 @@ clean_profiling:
 #	gcc -std=c99 -D xxx -pg ../src/operators/*.c ../src/trace.c ../src/utils.c ../src/inference.c ../src/pb/onnx.pb-c.c -o gprof tests.c -I/usr/local/include -L/usr/local/lib -lcunit -lprotobuf-c
 #	./gprof $(ts) $(tc)
 
+.phony:format
+HELP_format=run uncrustify to format code
+TARGET+=format
+format:
+	git ls-files -ico $(foreach PATTERN, $(FORMAT),-x '$(PATTERN)' ) | uncrustify -c ./.uncrustify.cfg -F - --no-backup --replace
+
+.phony:format-check
+HELP_format-check=check if code needs formatting and show diffs
+TARGET+=format-check
+format-check:
+	git ls-files -ico $(foreach PATTERN, $(FORMAT),-x '$(PATTERN)' ) \
+	| xargs -I % sh -c \
+	"uncrustify -c ./.uncrustify.cfg -f % \
+	 | git -c 'color.diff.new=normal 22' -c 'color.diff.old=normal 88' diff --exit-code --no-index --color --word-diff=color % -"
+
+.phony:onnx_generator
+HELP_onnx_generator=generate various onnx sources and headers
+TARGET+=onnx_generator
+onnx_generator:
+	cd scripts; python -m onnx_generator \
+	$(if $(ONNX_CUSTOM), --onnx $(abspath $(ONNX_CUSTOM))) \
+	$(if $(ONNX_INCLUDE), --include $(ONNX_INCLUDE)) \
+	$(if $(ONNX_EXCLUDE), --exclude $(ONNX_EXCLUDE)) \
+	$(if $(ONNX_VERSION), --version $(ONNX_VERSION)) \
+	$(if $(ONNX_DOMAINS), --domains $(ONNX_DOMAINS)) \
+	-vv \
+	--force \
+	$(shell git rev-parse --show-toplevel)
```

Comment:
Not working on macOS. Is there a way we can make it compatible?

Reply:
@alrevuelta 
is this still relevant?

Reply:
I'm asking because you merged